Tools/WinMLRunner/src/BindingUtilities.h

```diff
@@ -300,142 +326,161 @@ namespace BindingUtilities
-            return TensorValue::CreateFromArray(binding.GetShapeBuffer(), binding.GetDataBuffer());
-        }
-        else if (args.IsGarbageInput())
-        {
-            std::vector<int64_t> vecShape = {};
-            auto tensorDescriptorShape = tensorDescriptor.Shape();
-            for (UINT dim = 0; dim < tensorDescriptorShape.Size(); dim++)
+            INT64 dimSize = tensorShape.GetAt(dim);
+            if (dimSize > 0) // If the dimension is greater than 0, then it is known.
+            {
+                vecShape.push_back(dimSize);
+            }
+            else // otherwise, make sure that the dimension is -1, representing free dimension. If not, then it's an
+                 // invalid model.
             {
-                INT64 dimSize = tensorDescriptorShape.GetAt(dim);
-                if (dimSize > 0) // If the dimension is greater than 0, then it is known.
+                if (dimSize == -1)
                 {
-                    vecShape.push_back(dimSize);
+                    vecShape.push_back(1);
                 }
-                else // otherwise, make sure that the dimension is -1, representing free dimension. If not, then it's an
-                     // invalid model.
+                else
                 {
-                    if (dimSize == -1)
-                    {
-                        vecShape.push_back(1);
-                    }
-                    else
-                    {
-                        throw hresult_invalid_argument(L"Failed to create a tensor with an unknown dimension of: " +
-                                                       dimSize);
-                    }
+                    throw hresult_invalid_argument(L"Failed to create a tensor with an unknown dimension of: " +
+                                                   dimSize);
                 }
             }
-            auto tensorValue = TensorValue::Create(vecShape);
+        }
+        auto tensorValue = TensorValue::Create(vecShape);
 
-            com_ptr<ITensorNative> spTensorValueNative;
-            tensorValue.as(spTensorValueNative);
+        com_ptr<ITensorNative> spTensorValueNative;
+        tensorValue.as(spTensorValueNative);
 
-            BYTE* actualData;
-            uint32_t actualSizeInBytes;
-            spTensorValueNative->GetBuffer(
-                &actualData, &actualSizeInBytes); // Need to GetBuffer to have CPU memory backing tensorValue
+        BYTE* actualData;
+        uint32_t actualSizeInBytes;
+        spTensorValueNative->GetBuffer(&actualData,
+                                       &actualSizeInBytes); // Need to GetBuffer to have CPU memory backing tensorValue
+
+        if (!args.CsvPath().empty())
+        {
+            WriteDataToBinding<DataType>(tensorStringInput, reinterpret_cast<DataType*>(actualData), actualSizeInBytes);
+        }
+        else if (args.IsGarbageInput())
+        {
             return tensorValue;
         }
```

Comment:
remove this else if and add else if (args.IsImage()) { ... } return tensorValue;